### ISSUE TITLE
empty vector has no length?

### ISSUE THREAD

**Issue by gordonwoodhull** [att/rserve-js#4]
a non-empty vector comes through as an array plus `{r_type, r_attributes}`, and the extra metadata gets ignored because it looks like an array.

but an empty vector comes through as just those attributes, with no length, so it looks like an object.

this caused https://github.com/att/rcloud/issues/1663, which was easy enough to fix, but this is worth looking into

**Comment by s-u** [comment]
Yes, that seems like a bug - it should be simply an array of length 0 in JS so `[]`.
